src/module.c

```diff
@@ -445,8 +441,21 @@ int DropIndexCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     keepDocs = 1;
   }
 
-  RedisSearchCtx sctx = SEARCH_CTX_STATIC(ctx, sp);
-  Redis_DropIndex(&sctx, (delDocs || sp->flags & Index_Temporary) && !keepDocs);
+  if((delDocs || sp->flags & Index_Temporary) && !keepDocs) {
+    // We take a strong reference to the index, so it will not be freed
+    // and we can still use it to delete the keys
+    StrongRef oun_ref = StrongRef_Clone(global_ref);
```

Comment:
do we still need to take it under the assumption of main thread?

Reply:
yes, the "remove from globals" will release the global reference and the spec could be freed there, and we still need its doc table. I improved the comments.